client/rest/src/main/java/org/elasticsearch/client/ResponseException.java

```diff
@@ -34,9 +38,19 @@
         this.response = response;
     }
 
-    private static String buildMessage(Response response) {
-        return response.getRequestLine().getMethod() + " " + response.getHost() + response.getRequestLine().getUri()
+    private static String buildMessage(Response response) throws IOException {
+        String message = response.getRequestLine().getMethod() + " " + response.getHost() + response.getRequestLine().getUri()
                 + ": " + response.getStatusLine().toString();
+
+        HttpEntity entity = response.getEntity();
+        if (entity != null) {
+            if (entity.isRepeatable() == false) {
+                entity = new BufferedHttpEntity(entity);
```

Comment:
I think you need to return the BufferedHttpEntity in the response for this to work. Building it here will consume the underlying entity, but you still return that entity. So it'll be empty. I'm honestly not sure why the tests pass.